python/cudf/bindings/binops.pyx

```diff
@@ -200,47 +243,71 @@ cdef apply_compiled_op(gdf_column* c_lhs, gdf_column* c_rhs, gdf_column* c_out,
 
 def apply_op(lhs, rhs, out, op):
     """
-    Dispatches a binary op call to the appropriate libcudf function
+    Dispatches a binary op call to the appropriate libcudf function:
     """
-    check_gdf_compatibility(lhs)
-    check_gdf_compatibility(rhs)
     check_gdf_compatibility(out)
-
-    cdef gdf_column* c_lhs = column_view_from_column(lhs)
-    cdef gdf_column* c_rhs = column_view_from_column(rhs)
+    cdef gdf_column* c_lhs = NULL
+    cdef gdf_column* c_rhs = NULL
+    cdef gdf_scalar* c_scalar = NULL
```

Comment:
You don't need a pointer to a `gdf_scalar`. You can just make it trivially and skip the `free`. But it's your choice really.

Reply:
That's actually how I originally had it since stack memory tends to be faster, but I changed it to be consistent with the way columns are initialized